src/haveServerDataSelectors.js

```diff
@@ -119,9 +115,16 @@ export const getHaveServerData = (state: PerAccountState): boolean => {
     return false;
   }
 
+  const { zulipVersion } = getAccount(state);
+
+  // The server version comes with the rest of the server data.
+  if (!zulipVersion) {
+    return false;
+  }
+
   // We may have server data, but it would be from an ancient server that we
   // don't support, so it might be malformed.
-  if (!getServerVersion(state).isAtLeast(kMinAllowedServerVersion)) {
```

Comment:
nit:

    Oops: getHaveServerData was not an OK place to call
    getServerVersion. It has the following invariant:
    
      // This invariant will hold as long as we only call this function in a
      // context where we have server data.
      //
      // TODO(#5006): […]
      invariant(zulipVersion, 'zulipVersion must be non-null');

I think more to the point than that implementation comment is the jsdoc:

```
 * This function assumes we have server data for this account, and if not it
 * may throw.  […]
```